beacon-chain/p2p/subnets.go

```diff
@@ -585,3 +638,16 @@ func byteCount(bitCount int) int {
 	}
 	return numOfBytes
 }
+
+// interesect intersects two maps and returns a new map containing only the keys
+// that are present in both maps.
+func intersect(left map[uint64]int, right map[uint64]bool) map[uint64]bool {
+	result := make(map[uint64]bool, len(left))
```

Comment:
Not sure about giving a capacity hint here. If there is no intersection, then you've allocated space that won't be used. 

Do you expect that there will often be a complete overlap? If so, I'd take the minimum of right or left.


```suggestion
	result := make(map[uint64]bool, min(len(left), len(right)))
```

Reply:
>  If there is no intersection, then you've allocated space that won't be used.

Yes I agree. I wasn't sure about the best heuristic to use for the capacity hint.
I like your `min` idea, thanks!